vnext/Microsoft.ReactNative/Views/TextViewManager.cpp

```diff
@@ -274,23 +307,33 @@ YGMeasureFunc TextViewManager::GetYogaCustomMeasureFunc() const {
   return DefaultYogaSelfMeasureFunc;
 }
 
-void TextViewManager::OnDescendantTextPropertyChanged(ShadowNodeBase *node) {
-  if (auto element = node->GetView().try_as<xaml::Controls::TextBlock>()) {
-    // If name is set, it's controlled by accessibilityLabel, and it's already
-    // handled in FrameworkElementViewManager. Here it only handles when name is
-    // not set.
-    if (xaml::Automation::AutomationProperties::GetLiveSetting(element) != winrt::AutomationLiveSetting::Off &&
-        xaml::Automation::AutomationProperties::GetName(element).empty() &&
-        xaml::Automation::AutomationProperties::GetAccessibilityView(element) != winrt::Peers::AccessibilityView::Raw) {
-      if (auto peer = xaml::Automation::Peers::FrameworkElementAutomationPeer::FromElement(element)) {
-        peer.RaiseAutomationEvent(winrt::AutomationEvents::LiveRegionChanged);
+void TextViewManager::OnDescendantTextPropertyChanged(ShadowNodeBase *node, PropertyChangeType propertyChangeType) {
+  if (std::wcscmp(node->GetViewManager()->GetName(), GetName()) == 0) {
```

Comment:
curious if we would ever want to have text functionality exposed through the abi so that community VMs can reuse it, we need to come up with a better check than "what's your VM's name".

Maybe abstract this check away in a helper `IsTextNode` and add a comment that for the time being only the `TextViewManager` can use this functionality?

Reply:
So for Android Paper UIManager and how this was set up in v0.59 and earlier for RNW, we just used the managed type system to check if the shadow node was a particular type.

Presumably if we're going to expose core VMs through the ABI, we'll also need to expose shadow node functionality. I'm guessing we don't want to introduce any dynamic_cast / RTTI, so when we take this approach, is there a way for us to add type traits to each shadow node so we can have a non-dynamic cast?

Reply:
we wouldn't use rtti/dynamic_cast since that's not ABI safe, but instead we'd use a WinRT interface you can QueryInterface / `.as<T>()` for

Reply:
Added `ShadowNodeTypeUtils.h` to keep this logic in one place.